packages/content/src/attributes/types/birth/BirthDate.ts

```diff
@@ -37,6 +39,36 @@ export class BirthDate extends AbstractComplexValue implements IBirthDate {
     @validate()
     public year: BirthYear;
 
+    protected static override postFrom<T extends Serializable>(value: T): T {
+        if (!(value instanceof BirthDate)) {
+            throw new ConsumptionError("An unexpected error has occured with a BirthDate.");
+        }
+
+        if ((value.month.value === 2 && value.day.value === 30) || ([2, 4, 6, 9, 11].includes(value.month.value) && value.day.value === 31)) {
+            throw new ValidationError(
+                BirthDate.name,
+                nameof<BirthDate>((x) => x.day),
+                "The BirthDate is not a valid date. The chosen day does not exist in the chosen month."
+            );
+        }
+
+        const valueIsInLeapYear = DateTime.fromObject({
+            day: value.day.value,
+            month: value.month.value,
+            year: value.year.value
+        }).isInLeapYear;
+
+        if (value.month.value === 2 && !valueIsInLeapYear && value.day.value === 29) {
+            throw new ValidationError(
+                BirthDate.name,
+                nameof<BirthDate>((x) => x.year),
+                "The BirthDate is not a valid date. The 29 February only exists in leap years."
+            );
+        }
```

Comment:
You could just construct a luxon `DateTime` object and check if its valid.

```ts
import { DateTime } from "luxon";

const invalid = DateTime.fromObject({ year: 2023, month: 2, day: 29 });
console.log(invalid.isValid); // false

const valid = DateTime.fromObject({ year: 2024, month: 2, day: 29 });
console.log(valid.isValid); // true
```

This `.isValid` will tell you clearly whether a day is valid or not and luxon will catch any other errors unrelated to leap years, too. I understand you want to manually check for different cases to return nice errors, but I strongly suspect this to be overengineering. If anybody enters an invalid date, a simple message that the date is, in fact, invalid will help the user enough to rectify the errors.

The main use case for errors will be due to people mixing up the year, day and month fields when constructing the error. People messing up leap year dates is rather a fringe case, so no use putting too much effort into it. although I don't have a problem with you doing it anyway. Just make sure you catch _all_ errors before taking care of individual edge cases.

Reply:
yeah - check my comment. That's exactly what I was saying. I'd definitely vote for checking the Date at the end using luxon and make sure you didn't missed sth.

Reply:
now the question is only 

a) *only* validation by luxon
b) do the things we know ourselves

I'd vote with a.

Reply:
I'd second that. Just check `.isValid` and return a generic error message such as "the date you entered is invalid".